### PR TITLE
Add some flexibility in how to request a merge.

### PR DESCRIPTION
**Pull request by jerjou**

### CODE REVIEW

File: webhook-app/webhooks.py [hunk 1]
```diff
@@ -31,23 +32,26 @@ def check_for_auto_merge_trigger(text):
...
 
-    # The comment must address @dpebot directly.
-    if not '@{}'.format(github_helper.github_user()) in text:
+    # The comment must address @dpebot directly, on the same line
+    comment = re.search(
+        r'@{}\s+\b(.+)'.format(github_helper.github_user()), text)
+    if not comment:
         return False
+    else:
+        # Just get the meat of the command
```
Comment (jonparrott) [inline webhook-app/webhooks.py]: Mmm meat.

File: webhook-app/webhooks.py [hunk 2]
```diff
@@ -31,23 +32,26 @@ def check_for_auto_merge_trigger(text):
...
+        comment = comment.group(1).strip()
 
     # These should all be lowercase.
+    satisfaction = r'\b(passes|green|approv(al|es)|happy|satisfied)'
```
Comment (jonparrott) [inline webhook-app/webhooks.py]: I can't get no?

File: webhook-app/webhooks.py [hunk 3]
```diff
@@ -31,23 +32,26 @@ def check_for_auto_merge_trigger(text):
...
+    # The comment must address @dpebot directly, on the same line
+    comment = re.search(
+        r'@{}\s+\b(.+)'.format(github_helper.github_user()), text)
+    if not comment:
         return False
+    else:
+        # Just get the meat of the command
+        comment = comment.group(1).strip()
 
     # These should all be lowercase.
```
Comment (jonparrott) [inline webhook-app/webhooks.py]: Use a case-insensitive regex and you can remove this comment.

File: webhook-app/webhooks.py [hunk 4]
```diff
@@ -31,23 +32,26 @@ def check_for_auto_merge_trigger(text):
...
+        r'{}.+{},.+{}'.format(ci_tool, satisfaction, merge_action),
+        'lgtm',
+    )
 
-    for trigger in triggers:
-        if trigger in text:
-            return True
-
-    return False
+    return any(re.search(trigger, comment) for trigger in triggers)
```
Comment (jonparrott) [inline webhook-app/webhooks.py]: Noice functional programming.

File: webhook-app/webhooks.py (hunk 3 above)
Comment (jerjou) [inline webhook-app/webhooks.py]: Made both regex searches case-insensitive, and removed the call to `.lower()`

**Comment by jerjou** [comment]
@dpebot would you please merge when Travis passes? I would be quite grateful.

**Comment by jerjou** [comment]
curses. Does @dpebot not keep its own house gardened?? Shoeless cobbler's children indeed...

**Comment by tswast** [comment]
Added @dpebot to this repo. We better not accidentally bring a burger-filled robot apocalypse.

**Comment by jonparrott** [comment]
If I had a vote for the way the world would end, that's my vote.

**Pull request merged**